homeassistant/components/cover/aladdin_connect.py

```diff
@@ -0,0 +1,117 @@
+"""
+Platform for the Aladdin Connect cover component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/cover.aladdin_connect/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.cover import (CoverDevice, PLATFORM_SCHEMA,
+                                            SUPPORT_OPEN, SUPPORT_CLOSE)
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, STATE_CLOSED,
+                                 STATE_OPENING, STATE_CLOSING, STATE_UNKNOWN,
+                                 STATE_OPEN)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['aladdin_connect==0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+NOTIFICATION_ID = 'aladdin_notification'
+NOTIFICATION_TITLE = 'Aladdin Connect Cover Setup'
+
+STATES_MAP = {
+    'open': STATE_OPEN,
+    'opening': STATE_OPENING,
+    'closed': STATE_CLOSED,
+    'closing': STATE_CLOSING,
+    'unknown': STATE_UNKNOWN
+}
+
+SUPPORTED_FEATURES = SUPPORT_OPEN | SUPPORT_CLOSE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Aladdin Connect platform."""
+    from aladdin_connect import AladdinConnectClient
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    acc = AladdinConnectClient(username, password)
+
+    try:
+        if not acc.login():
+            raise ValueError("Username or Password is incorrect")
+        add_devices(AladdinDevice(acc, door) for door in acc.get_doors())
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        _LOGGER.error("%s", ex)
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+
+
+class AladdinDevice(CoverDevice):
+    """Representation of Aladdin Connect cover."""
+
+    def __init__(self, acc, device):
+        """Initialize the cover."""
+        self._acc = acc
+        self._device_id = device['device_id']
+        self._number = device['door_number']
+        self._name = device['name']
+        self._status = STATES_MAP.get(device['status'], STATE_UNKNOWN)
```

Comment:
Set `None` if status is not found in the dict. That's the default if the key is missing in the dict.
```py
self._status = STATES_MAP.get(device['status'])
```